### PR TITLE
feat: Implement "Announce Event" email for group followers

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Implement "Announce Event" email for group followers 

required for https://github.com/fossasia/open-event-frontend/pull/7604

### CODE REVIEW

File: app/api/schema/events.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ def validate_timezone(self, data, original_data):
     is_featured = fields.Bool(default=False)
     is_promoted = fields.Bool(default=False)
     is_demoted = fields.Bool(default=False)
+    is_cfs_enabled = fields.Bool(default=False)
```
Comment (ranjitkshah) [inline app/api/schema/events.py:85]:
```suggestion
     is_announced = fields.Bool(default=False)
```

File: app/api/custom/groups.py [hunk 2]
```diff
@@ -1,17 +1,61 @@
...
+@to_event_id
+@jwt_required
+def announce_event(group_id, event_id):
+    group = Group.query.get_or_404(group_id)
+    event = Event.query.get_or_404(event_id)
+    if event.group_id != group.id:
+        raise ForbiddenError({'source': 'event_id'}, "Event does not belong to the group")
+    if group.user_id != current_user.id:
+        raise ForbiddenError(
+            {'source': 'user_id'}, "User have not permissions to announce event"
```
Comment (MeghalBisht) [inline app/api/custom/groups.py:32]:
```suggestion
            {'source': 'user_id'}, "User does not have permissions to announce event"
```

**Comment by MeghalBisht** [comment]
```
Hello,

We can’t wait to welcome you for event_name! We’re pretty confident that you’re going to love it.
Until then, take a look at the following event details to take full advantage of event_name.

Name of event: event_name
Name of group: vocal
Date: 21 August 2021
Location: Location Not Announced
Link to public event page: https://localhost:4200/e/e217ac24
Event description:

You’re getting this message because your Open Event account is connected to this email address. You can unsubscribe from similar emails from this group by unfollowing the group here.

Best regards,
Open Event Team.
```

@mariobehling Please check the announcement email message.

**Review by MeghalBisht — COMMENTED** [review]
Works!

File: migrations/versions/rev-2021-07-21-19:22:46-78f3c24b41cf_.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+"""
+
+from alembic import op
+import sqlalchemy as sa
+import sqlalchemy_utils
+
+
+# revision identifiers, used by Alembic.
+revision = '78f3c24b41cf'
+down_revision = '7ddfc1ae036c'
```
Comment (MeghalBisht) [inline migrations/versions/rev-2021-07-21-19:22:46-78f3c24b41cf_.py:16]:
```suggestion
down_revision = 'f52267133dcc'
```

File: migrations/versions/rev-2021-07-21-19:22:46-78f3c24b41cf_.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+"""empty message
+
+Revision ID: 78f3c24b41cf
+Revises: 7ddfc1ae036c
```
Comment (MeghalBisht) [inline migrations/versions/rev-2021-07-21-19:22:46-78f3c24b41cf_.py:4]:
```suggestion
Revises: f52267133dcc
```

**Comment by mariobehling** [comment]
Thank you. This is an event announcement email. So, I change the logic a bit.

---

Subject: Join "Event Name" from Group_Name on event_date

Hello,

Your group group_name_with_link has just scheduled "event_name"! We’re pretty confident that you’re going to love it.

Date: event_date
Time: event_time (event_timezone)
Location: Location Not Yet Announced
Link to public event page: public_event_link

Reserve your spot! (Link this to event page)

Event description: Event description here. Event description here. Event description here. Event description here. Event description here. Event description here. Event description here. Event description here. Event description here. 

Best regards,
app_name

You’re getting this message because you are a group member of group_name_with_link on app_name. You can unsubscribe from these emails from this group by unfollowing the group here_with_link.

**Review by MeghalBisht — DISMISSED** [review]

**Comment by sachinchauhan2889** [comment]
email looping will handle in next PR of https://github.com/fossasia/open-event-server/issues/8147

**Pull request merged**